# SOME DESCRIPTIVE TITLE.
# Copyright (C) 2001-2025, Python Software Foundation
# This file is distributed under the same license as the Python package.
# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
#
#, fuzzy
msgid ""
msgstr ""
"Project-Id-Version: Python 3.9\n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: 2025-07-11 17:51+0000\n"
"PO-Revision-Date: 2017-02-16 17:48+0000\n"
"Language-Team: Portuguese (Brazil) (https://app.transifex.com/python-doc/"
"teams/5390/pt_BR/)\n"
"Language: pt_BR\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Plural-Forms: nplurals=3; plural=(n == 0 || n == 1) ? 0 : n != 0 && n % "
"1000000 == 0 ? 1 : 2;\n"

#: ../../library/asyncio-dev.rst:7
msgid "Developing with asyncio"
msgstr "Desenvolvendo com asyncio"

#: ../../library/asyncio-dev.rst:9
msgid ""
"Asynchronous programming is different from classic \"sequential\" "
"programming."
msgstr ""
"Асинхронне програмування відрізняється від класичного \"послідовного\" "
"програмування."

#: ../../library/asyncio-dev.rst:12
msgid ""
"This page lists common mistakes and traps and explains how to avoid them."
msgstr ""
"На цій сторінці наведено типові помилки та пастки та пояснено, як їх "
"уникнути."

#: ../../library/asyncio-dev.rst:19
msgid "Debug Mode"
msgstr "Modo de Depuração"

#: ../../library/asyncio-dev.rst:21
msgid ""
"By default asyncio runs in production mode.  In order to ease the "
"development asyncio has a *debug mode*."
msgstr ""
"За замовчуванням asyncio працює у робочому режимі. Щоб полегшити розробку, "
"asyncio має *режим налагодження*."

#: ../../library/asyncio-dev.rst:24
msgid "There are several ways to enable asyncio debug mode:"
msgstr "Є кілька способів увімкнути асинхронний режим налагодження:"

#: ../../library/asyncio-dev.rst:26
msgid "Setting the :envvar:`PYTHONASYNCIODEBUG` environment variable to ``1``."
msgstr "Встановлення змінної середовища :envvar:`PYTHONASYNCIODEBUG` на ``1``."

#: ../../library/asyncio-dev.rst:28
msgid "Using the :ref:`Python Development Mode <devmode>`."
msgstr "Usando o :ref:`Modo de Desenvolvimento do Python <devmode>`."

#: ../../library/asyncio-dev.rst:30
msgid "Passing ``debug=True`` to :func:`asyncio.run`."
msgstr "Passando ``debug=True`` para :func:`asyncio.run`."

#: ../../library/asyncio-dev.rst:32
msgid "Calling :meth:`loop.set_debug`."
msgstr "Chamando :meth:`loop.set_debug`."

#: ../../library/asyncio-dev.rst:34
msgid "In addition to enabling the debug mode, consider also:"
msgstr "Além de habilitar o modo de depuração, considere também:"

#: ../../library/asyncio-dev.rst:36
msgid ""
"setting the log level of the :ref:`asyncio logger <asyncio-logger>` to :py:"
"data:`logging.DEBUG`, for example the following snippet of code can be run "
"at startup of the application::"
msgstr ""

#: ../../library/asyncio-dev.rst:42
msgid ""
"configuring the :mod:`warnings` module to display :exc:`ResourceWarning` "
"warnings.  One way of doing that is by using the :option:`-W` ``default`` "
"command line option."
msgstr ""
"налаштування модуля :mod:`warnings` для відображення попереджень :exc:"
"`ResourceWarning`. Один із способів зробити це — скористатися параметром "
"командного рядка :option:`-W` ``default``."

#: ../../library/asyncio-dev.rst:47
msgid "When the debug mode is enabled:"
msgstr "Quando o modo de depuração está habilitado:"

#: ../../library/asyncio-dev.rst:49
msgid ""
"asyncio checks for :ref:`coroutines that were not awaited <asyncio-coroutine-"
"not-scheduled>` and logs them; this mitigates the \"forgotten await\" "
"pitfall."
msgstr ""
"asyncio перевіряє :ref:`співпрограми, які не були очікувані <asyncio-"
"coroutine-not-scheduled>`, і записує їх у журнал; це пом'якшує помилку "
"\"забутого очікування\"."

#: ../../library/asyncio-dev.rst:53
msgid ""
"Many non-threadsafe asyncio APIs (such as :meth:`loop.call_soon` and :meth:"
"`loop.call_at` methods) raise an exception if they are called from a wrong "
"thread."
msgstr ""
"Багато небезпечних для потоків асинхронних API (таких як методи :meth:`loop."
"call_soon` і :meth:`loop.call_at`) викликають виняток, якщо вони "
"викликаються з неправильного потоку."

#: ../../library/asyncio-dev.rst:57
msgid ""
"The execution time of the I/O selector is logged if it takes too long to "
"perform an I/O operation."
msgstr ""
"O tempo de execução de um seletor de E/S é registrado se demorar muito para "
"executar a operação E/S."

#: ../../library/asyncio-dev.rst:60
msgid ""
"Callbacks taking longer than 100ms are logged.  The :attr:`loop."
"slow_callback_duration` attribute can be used to set the minimum execution "
"duration in seconds that is considered \"slow\"."
msgstr ""

#: ../../library/asyncio-dev.rst:68
msgid "Concurrency and Multithreading"
msgstr "Concorrência e Múltiplas Threads"

#: ../../library/asyncio-dev.rst:70
msgid ""
"An event loop runs in a thread (typically the main thread) and executes all "
"callbacks and Tasks in its thread.  While a Task is running in the event "
"loop, no other Tasks can run in the same thread.  When a Task executes an "
"``await`` expression, the running Task gets suspended, and the event loop "
"executes the next Task."
msgstr ""
"Цикл подій виконується в потоці (зазвичай головному) і виконує всі зворотні "
"виклики та завдання у своєму потоці. Поки Завдання виконується в циклі "
"подій, жодні інші Завдання не можуть виконуватися в тому самому потоці. Коли "
"Завдання виконує вираз ``очікування``, запущене Завдання призупиняється, а "
"цикл подій виконує наступне Завдання."

#: ../../library/asyncio-dev.rst:76
msgid ""
"To schedule a :term:`callback` from another OS thread, the :meth:`loop."
"call_soon_threadsafe` method should be used. Example::"
msgstr ""
"Para agendar uma :term:`callback` de outra thread do SO, o método :meth:"
"`loop.call_soon_threadsafe` deve ser usado. Exemplo::"

#: ../../library/asyncio-dev.rst:81
msgid ""
"Almost all asyncio objects are not thread safe, which is typically not a "
"problem unless there is code that works with them from outside of a Task or "
"a callback.  If there's a need for such code to call a low-level asyncio "
"API, the :meth:`loop.call_soon_threadsafe` method should be used, e.g.::"
msgstr ""
"Майже всі асинхронні об’єкти не є потокобезпечними, що зазвичай не є "
"проблемою, якщо немає коду, який працює з ними поза Завданням або зворотним "
"викликом. Якщо існує потреба в такому коді для виклику низькорівневого "
"асинхронного API, слід використовувати метод :meth:`loop."
"call_soon_threadsafe`, наприклад::"

#: ../../library/asyncio-dev.rst:89
msgid ""
"To schedule a coroutine object from a different OS thread, the :func:"
"`run_coroutine_threadsafe` function should be used. It returns a :class:"
"`concurrent.futures.Future` to access the result::"
msgstr ""
"Щоб запланувати об’єкт співпрограми з іншого потоку ОС, слід використати "
"функцію :func:`run_coroutine_threadsafe`. Він повертає :class:`concurrent."
"futures.Future` для доступу до результату::"

#: ../../library/asyncio-dev.rst:102
msgid ""
"To handle signals and to execute subprocesses, the event loop must be run in "
"the main thread."
msgstr ""
"Para tratar sinais e executar subprocessos, o laço de eventos precisa "
"executar na thread principal."

#: ../../library/asyncio-dev.rst:105
msgid ""
"The :meth:`loop.run_in_executor` method can be used with a :class:"
"`concurrent.futures.ThreadPoolExecutor` to execute blocking code in a "
"different OS thread without blocking the OS thread that the event loop runs "
"in."
msgstr ""
"Метод :meth:`loop.run_in_executor` можна використовувати з :class:"
"`concurrent.futures.ThreadPoolExecutor` для виконання коду блокування в "
"іншому потоці ОС, не блокуючи потік ОС, у якому виконується цикл подій."

#: ../../library/asyncio-dev.rst:110
msgid ""
"There is currently no way to schedule coroutines or callbacks directly from "
"a different process (such as one started with :mod:`multiprocessing`). The :"
"ref:`Event Loop Methods <asyncio-event-loop>` section lists APIs that can "
"read from pipes and watch file descriptors without blocking the event loop. "
"In addition, asyncio's :ref:`Subprocess <asyncio-subprocess>` APIs provide a "
"way to start a process and communicate with it from the event loop. Lastly, "
"the aforementioned :meth:`loop.run_in_executor` method can also be used with "
"a :class:`concurrent.futures.ProcessPoolExecutor` to execute code in a "
"different process."
msgstr ""

#: ../../library/asyncio-dev.rst:124
msgid "Running Blocking Code"
msgstr "Executando código bloqueante"

#: ../../library/asyncio-dev.rst:126
msgid ""
"Blocking (CPU-bound) code should not be called directly.  For example, if a "
"function performs a CPU-intensive calculation for 1 second, all concurrent "
"asyncio Tasks and IO operations would be delayed by 1 second."
msgstr ""
"Код блокування (прив’язаний до процесора) не слід викликати безпосередньо. "
"Наприклад, якщо функція виконує інтенсивне обчислення ЦП протягом 1 секунди, "
"усі одночасні асинхронні завдання та операції введення-виведення будуть "
"відкладені на 1 секунду."

#: ../../library/asyncio-dev.rst:131
msgid ""
"An executor can be used to run a task in a different thread or even in a "
"different process to avoid blocking the OS thread with the event loop.  See "
"the :meth:`loop.run_in_executor` method for more details."
msgstr ""
"Виконавець можна використовувати для запуску завдання в іншому потоці або "
"навіть в іншому процесі, щоб уникнути блокування потоку ОС за допомогою "
"циклу подій. Додаткову інформацію див. у методі :meth:`loop.run_in_executor`."

#: ../../library/asyncio-dev.rst:140
msgid "Logging"
msgstr "Gerando logs"

#: ../../library/asyncio-dev.rst:142
msgid ""
"asyncio uses the :mod:`logging` module and all logging is performed via the "
"``\"asyncio\"`` logger."
msgstr ""
"asyncio usa o módulo :mod:`logging` e todo registro é feito via o "
"registrador ``\"asyncio\"``."

#: ../../library/asyncio-dev.rst:145
msgid ""
"The default log level is :py:data:`logging.INFO`, which can be easily "
"adjusted::"
msgstr ""

#: ../../library/asyncio-dev.rst:154
msgid "Detect never-awaited coroutines"
msgstr "Виявляти ніколи не очікувані співпрограми"

#: ../../library/asyncio-dev.rst:156
msgid ""
"When a coroutine function is called, but not awaited (e.g. ``coro()`` "
"instead of ``await coro()``) or the coroutine is not scheduled with :meth:"
"`asyncio.create_task`, asyncio will emit a :exc:`RuntimeWarning`::"
msgstr ""
"Коли функція співпрограми викликається, але не очікується (наприклад, "
"``coro()`` замість ``await coro()``) або співпрограма не запланована за "
"допомогою :meth:`asyncio.create_task`, asyncio видасть :exc:"
"`RuntimeWarning`::"

#: ../../library/asyncio-dev.rst:171 ../../library/asyncio-dev.rst:216
msgid "Output::"
msgstr "Saída::"

#: ../../library/asyncio-dev.rst:176 ../../library/asyncio-dev.rst:232
msgid "Output in debug mode::"
msgstr "Вивід у режимі налагодження::"

#: ../../library/asyncio-dev.rst:189
msgid ""
"The usual fix is to either await the coroutine or call the :meth:`asyncio."
"create_task` function::"
msgstr ""
"Звичайним виправленням є або очікування співпрограми, або виклик функції :"
"meth:`asyncio.create_task`::"

#: ../../library/asyncio-dev.rst:197
msgid "Detect never-retrieved exceptions"
msgstr "Виявлення ніколи не отриманих винятків"

#: ../../library/asyncio-dev.rst:199
msgid ""
"If a :meth:`Future.set_exception` is called but the Future object is never "
"awaited on, the exception would never be propagated to the user code.  In "
"this case, asyncio would emit a log message when the Future object is "
"garbage collected."
msgstr ""
"Якщо викликається :meth:`Future.set_exception`, але об’єкт Future ніколи не "
"очікується, виняток ніколи не поширюватиметься на код користувача. У цьому "
"випадку asyncio створюватиме повідомлення журналу, коли об’єкт Future "
"збиратиме сміття."

#: ../../library/asyncio-dev.rst:204
msgid "Example of an unhandled exception::"
msgstr "Exemplo de uma exceção não tratada::"

#: ../../library/asyncio-dev.rst:227
msgid ""
":ref:`Enable the debug mode <asyncio-debug-mode>` to get the traceback where "
"the task was created::"
msgstr ""
":ref:`Увімкніть режим налагодження <asyncio-debug-mode>`, щоб отримати "
"відстеження місця створення завдання:"
